crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -112,16 +127,16 @@ impl BlobVec {
     /// [`BlobVec`]'s `item_layout`
     /// - the item that was stored in `*value` is left logically uninitialised/moved out of after
     /// calling this function, and as such should not be used or dropped by the caller.
-    pub unsafe fn replace_unchecked(&mut self, index: usize, value: *mut u8) {
+    pub unsafe fn replace_unchecked(&mut self, index: usize, value: OwningPtr<'_>) {
         debug_assert!(index < self.len());
-        let ptr = self.get_unchecked(index);
+        let ptr = self.get_unchecked_mut(index).inner();
```

Comment:
Could this be left as `let ptr = self.get_unchecked_mut(index)` to avoid having to recreate the `OwningPtr` for `drop()`?